pkg/controller/cincinnati/cincinnati_controller.go

```diff
@@ -0,0 +1,393 @@
+	if err := controllerutil.SetControllerReference(instance, deployment, r.scheme); err != nil {
+		return err
+	}
+
+	// Check if this deployment already exists
+	found := &appsv1.Deployment{}
+	err := r.client.Get(ctx, types.NamespacedName{Name: deployment.Name, Namespace: deployment.Namespace}, found)
+	if err != nil && errors.IsNotFound(err) {
+		reqLogger.Info("Creating Deployment", "Namespace", deployment.Namespace, "Name", deployment.Name)
+		err := r.client.Create(ctx, deployment)
+		if err != nil {
+			handleErr(reqLogger, &instance.Status, "CreateDeploymentFailed", err)
+			return err
+		}
+	} else if err != nil {
+		handleErr(reqLogger, &instance.Status, "GetDeploymentFailed", err)
+		return err
+	}
+
+	// found existing deployment; let's compare and update if needed. We'll make
+	// a copy of what we found, overwrite values with those that were just now
+	// generated from the spec, and compare the result with what was found.
+	updated := found.DeepCopy()
+	updated.Spec.Replicas = deployment.Spec.Replicas
+	updated.Spec.Selector = deployment.Spec.Selector
+	updated.Spec.Strategy = deployment.Spec.Strategy
+	if updated.Spec.Template.ObjectMeta.Labels == nil {
+		updated.Spec.Template.ObjectMeta.Labels = map[string]string{}
+	}
+	for key, value := range deployment.Spec.Template.ObjectMeta.Labels {
+		updated.Spec.Template.ObjectMeta.Labels[key] = value
+	}
+	updated.Spec.Template.Spec.Volumes = deployment.Spec.Template.Spec.Volumes
+	containers := updated.Spec.Template.Spec.Containers
+	for i := range containers {
+		var original corev1.Container
+		switch containers[i].Name {
+		case NameContainerGraphBuilder:
+			original = newGBContainer(instance, r.operandImage)
+		case NameContainerPolicyEngine:
+			original = newPEContainer(instance, r.operandImage)
+		default:
+			reqLogger.Info("encountered unexpected container in pod", "Container.Name", containers[i].Name)
+			continue
+		}
+		containers[i].Image = original.Image
+		containers[i].ImagePullPolicy = original.ImagePullPolicy
+		containers[i].Command = original.Command
+		containers[i].Args = original.Args
+		containers[i].Ports = original.Ports
+		containers[i].Env = original.Env
+
+		// Resources don't seem to work with DeepEqual, so we check equality
+		// manually.
+		if containers[i].Resources.Limits == nil {
+			containers[i].Resources.Limits = corev1.ResourceList{}
+		}
+		for k, origVal := range original.Resources.Limits {
+			value, ok := containers[i].Resources.Limits[k]
+			if !ok || !origVal.Equal(value) {
+				containers[i].Resources.Limits[k] = origVal
+			}
+		}
+		if containers[i].Resources.Requests == nil {
+			containers[i].Resources.Requests = corev1.ResourceList{}
+		}
+		for k, origVal := range original.Resources.Requests {
+			value, ok := containers[i].Resources.Requests[k]
+			if !ok || !origVal.Equal(value) {
+				containers[i].Resources.Requests[k] = origVal
+			}
+		}
+		containers[i].VolumeMounts = original.VolumeMounts
```

Comment:
Some reason these aren't up with the others like it?

Reply:
I think only because I went in the order that things were specified in the godocs. The `Resources` part above is part of the same flow of setting attributes, but required special care.